lib/screenplay/suppressed_predictions.ex

```diff
@@ -38,6 +41,76 @@ defmodule Screenplay.SuppressedPredictions do
     |> Repo.all()
   end
 
+  @doc """
+  Get all suppressed predictions modified for transit data
+  """
+  @spec get_all_suppressed_predictions_transit_data() :: [SuppressedPrediction.t()]
+  def get_all_suppressed_predictions_transit_data do
+    places = Places.get()
+
+    SuppressedPrediction
+    |> Repo.all()
+    |> Enum.map(fn
+      %SuppressedPrediction{
+        route_id: "Green",
+        location_id: location_id,
+        direction_id: direction_id
+      } ->
+        # Go through Green Line route and repopulate the appropriate branches with the unique direction id
+        Enum.find(places, fn place -> place.id == location_id end).routes
+        |> Enum.filter(fn route -> String.contains?(route, "Green") end)
+        |> Enum.map(fn route ->
+          if String.contains?(route, "Green") do
+            %{
+              stop_id: location_id,
+              route_id: route,
+              direction_id: direction_id
+            }
+          end
+        end)
+
+      %SuppressedPrediction{
+        route_id: "Silver",
+        location_id: location_id,
+        direction_id: direction_id
+      } ->
+        # Go through Silver Line route and repopulate the appropriate
+        # route numbers that exist in the @valid_silver_line_routes
+        Enum.find(places, fn place -> place.id == location_id end).screens
+        |> Enum.flat_map(fn
+          %Screenplay.Places.Place.PaEssScreen{routes: routes} -> routes
+          _ -> []
+        end)
+        |> Enum.filter(fn route ->
+          route.id in @valid_silver_line_routes and
+            route.direction_id == direction_id
+        end)
+        |> Enum.uniq_by(fn route -> {route.id, route.direction_id} end)
+        |> Enum.map(fn route ->
+          %{
+            stop_id: location_id,
+            route_id: route.id,
+            direction_id: direction_id
+          }
+        end)
+
+      # If not Green or Silver, just pass along the route
+      %SuppressedPrediction{
+        route_id: route_id,
+        location_id: location_id,
+        direction_id: direction_id
+      } ->
+        [
+          %{
+            stop_id: location_id,
+            route_id: route_id,
+            direction_id: direction_id
+          }
```

Comment:
There are actually two kinds of suppression flags here: stop-based, and terminal. We're kind of rolling them up in the database because each stop/line/direction can only be one or the other, but the handling of those by transit data will probably be quite different, so I'm guessing they'll want them split out somehow. Two options that come to mind: we could generate two separate lists, or we could specify a new value for the "flag" field mentioned above, which designates terminal suppression. We might want to check with transit data and see what would work best for them.

As for separating them out on our end, the data structure PR that's about to be merged should make that straightforward.